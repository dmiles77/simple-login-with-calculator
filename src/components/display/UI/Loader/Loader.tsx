import React from 'react';
import styles from './Loader.module.css';

const Loader: React.FC = () => {
	const { Loader } = styles;
	return <div className={Loader}></div>;
};

export default Loader;
