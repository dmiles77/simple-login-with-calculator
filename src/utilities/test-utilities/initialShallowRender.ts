import { ShallowWrapper } from 'enzyme';

// Used to quickly test basic shallow wrapper components for render success:
const initialShallowRender = (
	comp: ShallowWrapper,
	className?: string
): void => {
	expect(comp.length).toBe(1);
	expect(comp.length).not.toBe(2);
	if (className) expect(comp).toHaveClassName(`${className}`);
};

export default initialShallowRender;
