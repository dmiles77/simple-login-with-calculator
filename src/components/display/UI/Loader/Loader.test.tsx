import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Loader from './Loader';
import initialShallowRender from '../../../../utilities/test-utilities/initialShallowRender';

describe('Loader component', () => {
	let component: ShallowWrapper;
	beforeEach(() => (component = shallow(<Loader />)));

	it('should render without errors', () => {
		initialShallowRender(component, 'Loader');
	});
});
