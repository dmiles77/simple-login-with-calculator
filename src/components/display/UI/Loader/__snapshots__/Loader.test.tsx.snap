// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should render without errors 1`] = `
<div
  className="Loader"
/>
`;
